apmproxy/receiver.go

```diff
@@ -164,3 +159,20 @@ func (c *Client) handleIntakeV2Events() func(w http.ResponseWriter, r *http.Requ
 		}
 	}
 }
+
+// URL: http://server/register/transaction
+func (c *Client) handleTransactionRegistration() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		txnID := r.Header.Get("x-elastic-transaction-id")
+		traceID := r.Header.Get("x-elastic-trace-id")
```

Comment:
I think we should be receiving the full `traceparent` and optionally `tracestate` here. `traceparent` is needed for setting the parent ID, in case the Lambda invocation is not the trace root. `tracestate` is needed for setting the `transaction.sample_rate` field.

Reply:
I started with just sending a JSON payload with all the fields available during its creation however, it felt a bit too much to encode JSON in the agent and decode it again here. Does it make sense to get `parentID` and `sample_rate` as headers rather than in the request body or do you foresee that in the future we might need more data from the agent?

Reply:
If we were going to future-proof it, I think we would be better off just passing through the whole Lambda payload and letting the extension pull the headers out. That way the extension can also pull other information out for use in the transaction it creates (e.g. for `transaction.type` and `transaction.name`); and perhaps also to enrich transactions coming from agents.

The agent can still generate `transaction.id` and pass that through to the extension in a header -- though I do wonder if we could derive a `transaction.id` from the AWS request ID. The Lambda payload would just be passed through as-is, no additional encoding by the agent.

Reply:
I have updated the code to accept the fully encoded JSON for the partial transaction and in the current logic I append `transaction.{result,outcome}` to the JSON so all fields set by the agent are respected when creating a proxy txn.

Reply:
That looks good. If we're passing a partial transaction rather than the Lambda invocation payload, do we still need to send the transaction ID as a separate header? Can we pull the transaction ID out of the transaction when the register endpoint is called?

What do you think about specifying a "vnd" Content-Type (e.g. `application/vnd.elastic.apm.transaction+json`), to allow for changes later in case we do want to send and parse the Lambda payload in the extension?

Reply:
> If we're passing a partial transaction rather than the Lambda invocation payload, do we still need to send the transaction ID as a separate header? Can we pull the transaction ID out of the transaction when the register endpoint is called?

Yes, we can. I did it in the draft version before moving to header approach and somehow missed it here. I have fixed this.

> What do you think about specifying a "vnd" Content-Type (e.g. application/vnd.elastic.apm.transaction+json), to allow for changes later in case we do want to send and parse the Lambda payload in the extension?

This sounds like a good idea to me. I have added a check in the extension for `content-type` header.